transmission_test.go

```diff
@@ -0,0 +1,70 @@
+package nbd
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+func TestListenAndServeContextCancel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dir, _ := os.MkdirTemp("", "")
+
+	sockFile := filepath.Join(dir, "nbd.sock")
+
+	// Start the server
+	exited := make(chan any)
+	go func() {
+		lErr := ListenAndServe(ctx, "unix", sockFile, Export{})
+		if lErr != nil {
+			t.Errorf("ListenAndServe returned an error: %v", lErr)
+		}
+		close(exited)
```

Comment:
Make this `defer close(exited)` at the top of the function (guard against `ListenAndServe` panicing in a future change).

Reply:
Good point